src/assets/Generator.Shared/OperationHelpers.cs

```diff
@@ -0,0 +1,368 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// This implements the ARM scenarios for LROs. It is highly recommended to read the ARM spec prior to modifying this code:
+    /// https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations
+    /// Other reference documents include:
+    /// https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
+    /// https://github.com/Azure/adx-documentation-pr/blob/master/sdks/LRO/LRO_AzureSDK.md
+    /// </summary>
+    /// <typeparam name="T">The final result of the LRO.</typeparam>
+    internal class OperationHelpers<T>
+    {
+        public static TimeSpan DefaultPollingInterval { get; } = TimeSpan.FromSeconds(1);
+
+        private static readonly string[] s_failureStates = {"failed", "canceled"};
```

Comment:
Do we do s_ prefixes for static?

Reply:
This is just copy paste of the existing code which used to be in ArmOperationHelpers.  This is actually the class for all LRO, but for some reason was originally named ArmOperationHelpers.  This changes moves the common scenario into a non Arm OperationHelpers, which allows Arm to then have its own implementation

Reply:
Yeah, was more of "i don't honestly know what the coding convention is here" than a "go fix this".